src/main/java/example/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package example.service.impl;
+
+import example.dto.userDtos.UserRegistrationRequestDto;
+import example.dto.userDtos.UserResponseDto;
+import example.exception.RegistrationException;
+import example.mapper.UserMapper;
+import example.model.User;
+import example.repository.user.UserRepository;
+import example.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto request) throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(request.getEmail())) {
```

It's better, to avoid calling isPresent()